core/src/main/java/org/apache/iceberg/MetadataPaths.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Utility class that contains path conversion methods.
+ */
+public final class MetadataPaths {
+
+  private MetadataPaths() {
+  }
+
+  /**
+   * Convert a given relative path to absolute path for the table by appending the base table location
+   * @param path relative path to be converted
+   * @param tableLocation base table location
+   * @param properties table properties
+   * @return the absolute path
+   */
+  public static String toAbsolutePath(String path, String tableLocation, Map<String, String> properties) {
+    Preconditions.checkArgument(path != null && path.trim().length() > 0);
+    // TODO: Fix this after tests are changed to always pass the table location. Table location cannot be null.
+    if (tableLocation == null) {
+      return path;
+    }
+    // convert to absolute path by appending the table location
+    return useRelativePath(properties) && !path.startsWith(tableLocation) ? tableLocation + "/" + path : path;
```

Comment:
Shouldn't we add this method to the LocationProviders?
For me extending the LocationProvider with additional default implementation which handles relative patch would be more natural.

Reply:
IIUC, the `LocationProvider` class is for data files and is used to provide locations to write the files. In this PR, we do not wish to change where data files and/or manifest files are written. We just want to change the references to any paths inside `metadata` files to relative paths. This way, if a table is relocated/moved, we don't have to change a bunch of metadata files.

Reply:
You are right. Would it be better to introduce something similar for metadata file locations as well? We can have 2 implementations:
* AbsoluteMetadataLocationProvider
* RelativeMetadataLocationProvider

Or even a single implementation at this point handling both?
This could hide the complexity of calculating the path whenever it is needed and we can just pass down this provider for every metadata file (even encapsulate the io, so we do not have to pass that around too)